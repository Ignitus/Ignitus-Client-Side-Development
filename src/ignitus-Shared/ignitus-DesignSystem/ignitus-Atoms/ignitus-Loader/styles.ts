import styled from '@emotion/styled';
import { IgnitusLogo } from '../../ignitus-Assets/ignitus-Icons/ignitusLogo';
import { IgnitusBlue, GreyLight } from '../colors';

export const Spinner = styled.div`
  position: absolute;
  top: 0;
  left: 0;

  margin: auto;
  border: 1rem solid ${GreyLight};
  border-radius: 50%;
  border-top: 1rem solid ${IgnitusBlue};
  height: calc(100% - 2rem);
  width: calc(100% - 2rem);
  -webkit-animation: spin 2.5s linear infinite;
  animation: spin 3s linear infinite;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const Logo = styled(IgnitusLogo)`
  height: 65%;
  fill: ${IgnitusBlue};
`;

export const Container = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: auto;
  width: 10rem;
  height: 10rem;
`;
