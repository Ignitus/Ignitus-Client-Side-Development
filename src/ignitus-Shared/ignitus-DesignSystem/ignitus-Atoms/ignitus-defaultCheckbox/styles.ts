import styled from '@emotion/styled';
import { Paragraph } from '../typography';
import { LabelType } from './types';

export const CheckBox = styled.input`
  height: 1rem;
  width: 1rem;
  outline: none;
`;
export const Label = styled.label`
  margin-left: 0.4rem;
`;

export const StyledParagraph = styled(Paragraph)<LabelType>`
  font-size: ${props => props.size};
`;

export const Container = styled.div`
  display: inline-flex;
  align-items: center;
`;
