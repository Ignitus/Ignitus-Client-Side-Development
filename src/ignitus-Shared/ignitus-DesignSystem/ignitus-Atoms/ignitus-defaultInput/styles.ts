import styled from '@emotion/styled';
import * as C from '../colors';
import * as F from '../fonts';

export const Input = styled.input`
  height: ${props => props.height};
  width: ${props => props.width};
  background-color: ${C.GreyLight};
  padding-left: 2rem;
  color: ${C.GreySecondaryText};
  font-weight: ${F.SemiBold};
  font-size: ${F.XL};
  border: none;
  border-radius: 17px;
  outline: none;
  margin: 0.3rem 0;

  ::placeholder {
    color: ${C.GreySecondaryText};
    font-weight: ${F.SemiBold};
    font-size: ${F.MD};
    opacity: 0.5;
  }
`;
