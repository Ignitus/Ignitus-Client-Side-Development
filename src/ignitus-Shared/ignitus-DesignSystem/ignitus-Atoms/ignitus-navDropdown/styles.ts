import styled from '@emotion/styled/macro';
import { White, IgnitusBlue } from '../colors';
import { Icon as I } from '../../../ignitus-Utilities/Components/icon';
import { StyleProps } from './types';
import { MD } from '../fonts';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 12rem;
`;

export const Column = styled.div`
  background-color: ${White};
  border-radius: 1rem;
  margin-top: 1rem;
`;

export const NavigationHeading = styled.div`
  align-items: center;
  color: ${IgnitusBlue};
  cursor: pointer;
  display: flex;
  font-size: ${MD};
  justify-content: flex-end;
`;

export const Icon = styled(I)<StyleProps>`
  height: 2rem;
  fill: ${IgnitusBlue};
  margin: 0 0.5rem;
`;

export const Text = styled.div`
  text-align: center;
`;
