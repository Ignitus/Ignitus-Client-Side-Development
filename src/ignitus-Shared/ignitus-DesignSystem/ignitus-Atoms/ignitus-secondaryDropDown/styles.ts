import styled from '@emotion/styled';
import * as C from '../colors';
import * as F from '../fonts';

export const ParentContainer = styled.div`
  display: inline-flex;
  flex-direction: column;
`;

export const SearchBarWrapper = styled.div`
  width: 20rem;
  height: 2rem;
`;

export const SearchBar = styled.input`
  outline: none;
  border: 0.1rem solid ${C.IgnitusBlue};
  border-radius: 0.5rem;
  width: 20rem;
  height: 2rem;
  padding-left: 0.25rem;
  color: ${C.IgnitusBlue};
  font-weight: ${F.SemiBold};
  &::placeholder {
    color: ${C.IgnitusBlue};
    font-weight: ${F.SemiBold};
  }
`;

export const OptionsContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin: 1rem 0.25rem;
  background-color: ${C.White};
  overflow: auto;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.1);
  border-radius: 0.5rem;
`;

export const CardWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-items: baseline;
  padding: 0.5rem;
  height: 3rem;
  :hover {
    background-color: ${C.GreyLight};
  }
  flex: 1;
`;

export const Avatar = styled.img`
  border-radius: 50%;
  width: 2.5rem;
  margin: 0 1rem 0 1rem;
`;

export const NameWrapper = styled.div`
  margin: 0.5rem;
`;
