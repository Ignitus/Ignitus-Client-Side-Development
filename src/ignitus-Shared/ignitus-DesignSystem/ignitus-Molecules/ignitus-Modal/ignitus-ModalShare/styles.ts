import styled from '@emotion/styled';
import { Paragraph as P, Normal, MD, White, boxShadowColor } from '../../../..';
import { Icon } from '../../../../ignitus-Utilities/Components/icon';

import { flexibleColDiv, flexibleRowDiv } from '../../../shared';

export const ModalShare = styled.div`
  background-color: ${White};
  background-size: contain;
  border-radius: 1rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${boxShadowColor};
  width: 35rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  padding: 1.5rem;
  justify-content: space-between;
`;

export const BottomSection = styled(flexibleColDiv)`
  padding: 0.4rem 0.8rem 1.5rem 1.5rem;
  align-items: flex-start;
`;

export const IconsContainer = styled(flexibleRowDiv)`
  border-radius: 1rem;
  display: inline-flex;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const Paragraph = styled(P)`
  font-weight: ${Normal};
  font-size: ${MD};
  margin: 2rem 0.6rem 0.6rem 0.6rem;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;
