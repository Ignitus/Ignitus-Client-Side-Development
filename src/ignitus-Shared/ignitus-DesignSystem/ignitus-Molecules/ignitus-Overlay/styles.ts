import styled from '@emotion/styled';
import { flexibleColDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { Heading1 } from '../../..';

export const Overlay = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 102, 0.89);
  backdrop-filter: blur(10px);
  width: 100%;
  height: 100%;
`;

export const Logo = styled(Icon)`
  margin-bottom: 4rem;
  height: 7.5rem;
  path {
    fill: white;
  }
`;

export const Heading = styled(Heading1)`
  margin-bottom: 1.5rem;
`;

export const MessageSection = flexibleColDiv;
