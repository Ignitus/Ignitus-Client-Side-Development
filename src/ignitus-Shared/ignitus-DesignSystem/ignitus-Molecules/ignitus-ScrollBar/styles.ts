import styled from '@emotion/styled';

export const Scroll = styled.div`
  margin: 30px;
  float: left;
  height: 300px;
  width: 65px;
  background: #f5f5f5;
  overflow-y: scroll;
  margin-bottom: 25px;

  ::-webkit-scrollbar {
    width: 12px;
    background: #f5f5f5;
  }

  ::-webkit-scrollbar-track {
    box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #f5f5f5;
  }

  ::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: #f90;
  }
`;

export const Overflow = styled.div`
  min-height: 450px;
`;
