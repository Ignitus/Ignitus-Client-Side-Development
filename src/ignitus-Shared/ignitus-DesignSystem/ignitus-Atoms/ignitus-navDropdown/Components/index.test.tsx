import React from 'react';
import { mount, render } from 'enzyme';
import { DashboardNavigationDropdown } from './index';
import { Icon, NavigationHeading, Column } from '../styles';
import { AppIcon } from '../../../../types/iconsTypes/iconEnums';

const props = {
  Options: [{ title: 'title' }],
  Name: 'name',
};

it('should render', () => {
  const tree = render(<DashboardNavigationDropdown {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<DashboardNavigationDropdown />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<DashboardNavigationDropdown {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('it should render Name prop inside NavigationHeading surrounded by static icons', () => {
    const NavHeadingIcons = [AppIcon.StudentIcon, AppIcon.FilledArrowDownIcon];
    expect(
      wrapper
        .find(NavigationHeading)
        .children()
        .find(Icon).length,
    ).toBe(2);
    expect(wrapper.find(NavHeadingIcons[0]).length).toBe(1);
    expect(wrapper.find(NavHeadingIcons[1]).length).toBe(1);
    const RenderedNavHeadingIcons = [
      wrapper
        .find(NavigationHeading)
        .find(Icon)
        .at(0)
        .props().name,
      wrapper
        .find(NavigationHeading)
        .find(Icon)
        .at(1)
        .props().name,
    ];
    expect(RenderedNavHeadingIcons[0]).toBe(NavHeadingIcons[0]);
    expect(RenderedNavHeadingIcons[1]).toBe(NavHeadingIcons[1]);
    const NavigationHeadingTextWithIconNames = wrapper
      .find(NavigationHeading)
      .text() as string;

    // + 1 in slice is done due to wrapper rendering 'Student Icon' instead of 'StudentIcon'
    const NavigationHeadingText = NavigationHeadingTextWithIconNames.slice(
      RenderedNavHeadingIcons[0].length + 1,
    ).slice(0, -RenderedNavHeadingIcons[1].length);
    expect(NavigationHeadingText).toBe(props.Name);
  });

  it('it should render Options only after simulating click', () => {
    expect(wrapper.find(Column).length).toBe(0);
    wrapper.find(NavigationHeading).simulate('click');
    expect(wrapper.find(Column).length).toBe(props.Options.length);

    props.Options.forEach((option, index) => {
      expect(
        wrapper
          .find(Column)
          .at(index)
          .text()
          .trim(),
      ).toBe(option.title);
    });
  });

  it('should render given icon', () => {
    expect(wrapper.find(Icon).length).toBe(2);
    expect(wrapper.find(AppIcon.StudentIcon).length).toBe(1);
    expect(wrapper.find(AppIcon.FilledArrowDownIcon).length).toBe(1);
  });
});
