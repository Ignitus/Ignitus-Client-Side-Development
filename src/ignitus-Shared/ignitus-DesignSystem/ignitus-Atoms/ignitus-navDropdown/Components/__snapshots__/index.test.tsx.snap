// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  class="css-o5m53g-Container e4rsefi0"
>
  <div
    class="css-6yrc38-NavigationHeading e4rsefi2"
  >
    <svg
      aria-label="Student Icon"
      class="css-1kbib9z-Icon e4rsefi3"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Student Icon
      </title>
      <path
        d="M9 11.75a1.25 1.25 0 100 2.5 1.25 1.25 0 000-2.5zm6 0a1.25 1.25 0 100 2.5 1.25 1.25 0 000-2.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37a9.974 9.974 0 0010.41 3.97c.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8z"
        fill="#006"
      />
    </svg>
    name
    <svg
      aria-label="FilledArrowDownIcon"
      class="css-1kbib9z-Icon e4rsefi3"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        FilledArrowDownIcon
      </title>
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
      />
    </svg>
  </div>
</div>
`;
